src/transaction/txParser.ts

```diff
@@ -183,34 +225,58 @@ const parseTxCerts = (txCertificates: any[]): _Certificate[] => {
   )
 }
 
-const parseTxWithdrawals = (
-  withdrawals: any,
-): _Withdrawal[] => {
+const parseTxWithdrawals = (withdrawals: any): _Withdrawal[] => {
   if (!isWithdrawalsMap(withdrawals)) {
     throw Error(Errors.WithrawalsParseError)
   }
   return Array.from(withdrawals).map(([address, coins]): _Withdrawal => ({ address, coins: BigInt(coins) }))
 }
 
-const parseTtl = (ttl: any): number => {
-  if (ttl === null || ttl === undefined) {
-    throw Error(Errors.MssingTTLParseError)
+const parseFee = (fee: any): Lovelace => {
+  if (!isLovelace(fee)) {
+    throw Error(Errors.FeeParseError)
+  }
+  return BigInt(fee)
+}
+
+const parseTtl = (ttl: any): number | undefined => {
+  if (ttl && !Number.isInteger(ttl)) {
+    throw Error(Errors.TTLParseError)
   }
   return ttl
 }
 
+const parseValidityIntervalStart = (validityIntervalStart: any): number | undefined => {
+  if (validityIntervalStart && !Number.isInteger(validityIntervalStart)) {
+    throw Error(Errors.ValidityIntervalStartParseError)
+  }
+  return validityIntervalStart
+}
+
+const parseMetaDataHash = (metaDataHash: any): Buffer | undefined => {
+  if (metaDataHash && !Buffer.isBuffer(metaDataHash)) {
+    throw Error(Errors.MetaDataHashParseError)
+  }
+  return metaDataHash
+}
+
 const parseUnsignedTx = ([txBody, meta]: _UnsignedTxDecoded): _UnsignedTxParsed => {
+  if (txBody.get(TxBodyKeys.MINT)) {
+    throw Error(Errors.MintUnsupportedError)
+  }
   const inputs = parseTxInputs(txBody.get(TxBodyKeys.INPUTS))
   const outputs = parseTxOutputs(txBody.get(TxBodyKeys.OUTPUTS))
-  const fee = BigInt(txBody.get(TxBodyKeys.FEE))
+  const fee = parseFee(txBody.get(TxBodyKeys.FEE))
   const ttl = parseTtl(txBody.get(TxBodyKeys.TTL))
   const certificates = parseTxCerts(
     txBody.get(TxBodyKeys.CERTIFICATES) || [],
   )
   const withdrawals = parseTxWithdrawals(
     txBody.get(TxBodyKeys.WITHDRAWALS) || new Map(),
   )
-  const metaDataHash = txBody.get(TxBodyKeys.META_DATA_HASH) as Buffer
+  const metaDataHash = parseMetaDataHash(txBody.get(TxBodyKeys.META_DATA_HASH))
```

Comment:
just curious - do we currently support transactions with metadata? even for Trezor? because Trezor signs the whole tx with full metadata if I'm not mistaken

Reply:
I'm not sure how metadata works. This would need further effort and I suggest to solve this in another PR.